### ISSUE TITLE
Separate dynamic Inscription state from the Inscription resource

### ISSUE THREAD

**Issue by jack-linden** [hirosystems/ordinals-api#34]
The current Inscription resource contains a mixture of 
1. static genesis data
2. dynamic ordinal location data

One consideration that could be made to allow for optimal caching (by either consumer or server) would be to separate these API resources such that they could be queried for separately or together (i.e., included).

**Static fields**
* id
* number
* genesis_address
* genesis_block_height
* genesis_block_hash
* genesis_tx_id
* genesis_fee
* genesis_timestamp
* sat_ordinal
* sat_rarity
* sat_coinbase_height
* mime_type
* content_type
* content_length

**Dynamics fields**
* tx
* address
* location
* output
* value
* offset
* timestamp

**Proposed**
Inscription -> InscriptionLocation[] (one-to-many)

**Inscription**
* id
* number
* genesis_address
* genesis_block_height
* genesis_block_hash
* genesis_tx_id
* genesis_fee
* genesis_timestamp
* sat_ordinal
* sat_rarity
* sat_coinbase_height
* mime_type
* content_type
* content_length

**InscriptionLocation**
* id (auto-gen or composite of [inscription_id,location]
* inscription_id (FK to Inscription.id)
* address
* tx
* location
* output
* value
* offset
* timestamp

**API Endpoint impacts**
GET /inscriptions and GET /inscription/:id
* Expose additional query parameter "include" which can specify "inscription_location" to indicate side-loading of inscription locations
Example: GET /inscriptions?include=inscription_location
```
{
"inscriptions": [{},...,{}], 
"inscription_locations": [{},...,{}], 
"total" : 20,
"offset" : 0,
"limit": 60
}
```

Alternative (map based locations format)
```
{
"inscriptions": [{},...,{}], 
"inscription_locations": {
    "inscription_id_1: [{},{}],
    ...
    "inscription_id_n: [{},{}],
}, 
"total" : 20,
"offset" : 0,
"limit": 60
}
```
Alternative (embedded)
```
{
"inscriptions": [{
    "id": "xyzi0",
    "number" : 203,
     ...
     "inscription_locations": [{},...,{}]
},], 
"total" : 20,
"offset" : 0,
"limit": 60
}
```
Notes
* Slight deviation from Hiro's stylistic use of "results" as its top level data JSON field name. My preference is that the top-level data name field corresponds to the particular resource you're querying for.
* Also ties into https://github.com/hirosystems/ordinals-api/issues/25 where InscriptionLocation represents the indexed history of ordinal transfers

**Comment by rafaelcr** [comment]
Hi @jack-linden this is a very good idea and it's already being addressed as part of #23 ... we will add a new endpoint that gives you all the location history of a specific inscription (we already follow a DB schema very similar to the one you proposed, we were just working on the indexing part to get this right)

**Comment by rafaelcr** [comment]
Closing as duplicate of #23
